manifests/init.pp

```diff
@@ -278,6 +279,7 @@
   Boolean $ssl_erl_dist                                            = $rabbitmq::params::ssl_erl_dist,
   Optional[String] $rabbitmqadmin_package                          = $rabbitmq::params::rabbitmqadmin_package,
   Array $archive_options                                           = $rabbitmq::params::archive_options,
+  Optional[Array] $loopback_users                                  = $rabbitmq::params::loopback_users,
```

Comment:
This is a strange quirk of how Puppet works, but a param can't be Optional if it has a default. Well, technically, it can, but it can only be overridden with `undef` using hiera, so it's safer to not make it optional, and pass in an empty array (many people's preferred pattern anyway) to set it to an empty list when necessary.

Reply:
Yep, you're right, I'm working with puppet 4 and the automatic hiera bind then I forgot the "old" params pattern; I think an empty list as default value for this parameter is not the best choice to achieve the behaviour that you could expect, so I could make not optional this parameter a fix the default value to the list with the guest user, let's see

Reply:
Right, I think the right fix is to make the parameter not optional, and if users want to allow the guest user to login remotely as well, they can pass in an empty array. You could explicitly mention that in the param docs if you want to make it more clear.

Reply:
But the default value should be `['guest']` and then just `Array $loopback_users`. I think we're saying the same thing 😃

Reply:
Yes, we're in the same page